doc/source/exclude-patterns.md

```diff
@@ -0,0 +1,46 @@
+## Introduction
+
+The optional argument `exclude_patterns` in the `in_toto_run` API, also used by
+`--exclude` in the `in-toto-run` command line tool, applies
+[pathspec](http://python-path-specification.readthedocs.io) to compile
+[gitignore-style patterns](https://git-scm.com/docs/gitignore). Artifacts
+(materials and products) matched by an exclude pattern are not recorded when
+generating link metadata. If a pattern matches a directory, all files and
+subdirectories are also excluded.
+
+## Pattern Formats
+
+- Single asterisks match everything except a slash.
+- Question marks match any one character except a slash.
+- Ranges of characters can be used in patterns using the `[]` notation (i.e.,
+  `[a-z]`).
+- A forward slash indicates a directory separator. Separators at the beginning
+  or middle (or both) of a pattern are relative to the current directory.
+  Separators at the end of the pattern only match directories. (e.g.,
+  `doc/frotz/` matches `doc/frotz` directory but not `a/doc/frotz` directory;
+  however `frotz/` matches `frotz` and `a/frotz`).
```

Comment:
I don't understand how these examples relate to the rule "Separators at the end of the pattern only match directories".

And I don't understand why `frotz/` matches `a/frotz`. Wouldn't we need a pattern `**/frotz` for for that instead?

Reply:
```
>>> from pathspec import PathSpec 
>>> from in_toto.runlib import _apply_exclude_patterns 
>>> exclude_patterns = ["frotz/"] 
>>> exclude_filter = PathSpec.from_lines('gitwildmatch', exclude_patterns)
>>> _apply_exclude_patterns(["frotz/", "a/frotz/"], exclude_filter)
[]
>>> _apply_exclude_patterns(["frotz/", "a/frotz"], exclude_filter)
['a/frotz']
```

I've updated the document to read `a/frotz/` instead.